dockerclient/client.go

```diff
@@ -556,7 +556,7 @@ func randSeq(source string, n int) (string, error) {
 
 // LoadImage checks the client for an image matching from. If not found,
 // attempts to pull the image and then tries to inspect again.
-func (e *ClientExecutor) LoadImage(from string) (*docker.Image, error) {
+func (e *ClientExecutor) LoadImage(from string, platform string) (*docker.Image, error) {
```

Comment:
Breaking changes in Public API

Reply:
This could be handled by renaming the method to `LoadImageForPlatform()`, or something along those lines, and letting it take the platform spec as you're describing, defaulting to the current platform if it's passed an empty value (as that's what happens when go-dockerclient's `PullImageOptions.Platform` isn't set, and what the engine does when the API request it receives doesn't have one) and adding a new `LoadImage()`, with the same arguments and return values as the current one, that just calls `LoadImageForPlatform()` with an empty platform value.

Reply:
Thanks. Pushed a new version.